packages/zowe-explorer/src/dataset/actions.ts

```diff
@@ -701,36 +619,226 @@ async function handleUserSelection(newDSProperties, dsType): Promise<string> {
+            ZoweLogger.debug(localizedStrings.opCancelled);
+            api.Gui.showMessage(localizedStrings.opCancelled);
         }
     });
 
     // Show quickpick and store the user's input
     quickpick.show();
     const choice2 = await api.Gui.resolveQuickPick(quickpick);
-    const pattern = choice2.label;
     quickpick.dispose();
-
+    if (!choice2) {
+        return;
+    }
+    const pattern = choice2.label;
     const showPatternOptions = async (): Promise<void> => {
         const options: vscode.InputBoxOptions = {
-            value: newDSProperties.find((prop) => prop.label === pattern).value,
-            placeHolder: newDSProperties.find((prop) => prop.label === pattern).placeHolder,
+            value: newDSProperties?.find((prop) => prop.label === pattern)?.value,
+            placeHolder: newDSProperties?.find((prop) => prop.label === pattern)?.placeHolder,
         };
         newDSProperties.find((prop) => prop.label === pattern).value = await api.Gui.showInputBox(options);
     };
 
     if (pattern) {
         // Parse pattern for selected attribute
-        switch (pattern) {
-            case " + Allocate Data Set":
-                return new Promise((resolve) => resolve(` + Allocate Data Set`));
-            default:
-                await showPatternOptions();
-                break;
+        if (pattern === localizedStrings.allocString) {
+            return Promise.resolve(localizedStrings.allocString);
+        } else {
+            await showPatternOptions();
+        }
+        return Promise.resolve(handleUserSelection());
+    }
+}
+
+async function getDataSetName(): Promise<string> {
+    const options: vscode.InputBoxOptions = {
+        placeHolder: localize("createFile.inputBox.placeHolder", "Name of Data Set"),
+        ignoreFocusOut: true,
+        validateInput: (text) => {
+            return dsUtils.validateDataSetName(text) === true ? null : localize("dataset.validation", "Enter valid dataset name");
+        },
+    };
+    let dsName = await api.Gui.showInputBox(options);
+    if (!dsName) {
+        return;
+    }
+    dsName = dsName.trim().toUpperCase();
+    return dsName;
+}
+
+async function getDsTypeForCreation(datasetProvider: api.IZoweTree<api.IZoweDatasetTreeNode>): Promise<string> {
+    const stepTwoOptions: vscode.QuickPickOptions = {
+        placeHolder: localize("createFile.quickPickOption.dataSetType", "Template of Data Set to be Created"),
+        ignoreFocusOut: true,
+        canPickMany: false,
+    };
+    //get array of template names
+    const dsTemplateNames = getTemplateNames(datasetProvider);
+    const stepTwoChoices = [
+        ...dsTemplateNames,
+        localizedStrings.dsBinary,
+        localizedStrings.dsC,
+        localizedStrings.dsClassic,
+        localizedStrings.dsPartitioned,
+        localizedStrings.dsSequential,
+    ];
+    // eslint-disable-next-line no-return-await
+    return await api.Gui.showQuickPick(stepTwoChoices, stepTwoOptions);
```

Comment:
Could we potentially address this by making `getDsTypeForCreation` return a `Thenable<string>`? This way, users can still await the function, but we can keep the lint rule enabled for this line and it will avoid the need to `await` here as well.